assets/js/src/tracking.js

```diff
@@ -23,17 +17,19 @@ export const trackListProducts = ( {
 	products,
 	listName = __( 'Product List', 'woocommerce-google-analytics-integration' ),
 } ) => {
-	trackEvent( 'view_item_list', {
-		event_category: 'engagement',
-		event_label: __(
-			'Viewing products',
-			'woocommerce-google-analytics-integration'
-		),
-		items: products.map( ( product, index ) => ( {
-			...getProductImpressionObject( product, listName ),
-			list_position: index + 1,
-		} ) ),
-	} );
+	if ( products.length > 0 ) {
+		trackEvent( 'view_item_list', {
+			item_list_id: 'engagement',
```

Comment:
:scroll: :nail_care: 
If I understand the docs correctly, this id is to identify the list from which we're reporting.
With blocks, we can have multiple lists on different pages. But currently, we send `engagement` regardless of which list we track. Maybe we could somehow forward the block instance id, or at least page ID?

Reply:
That would be a nice addition! I've opened issue #324 so as to not get off track in this PR.